internal/certprovider.go

```diff
@@ -1,24 +1,60 @@
 package internal
 
 import (
+	"archive/zip"
+	"bytes"
+	"crypto/sha256"
 	"crypto/x509"
 	_ "embed"
+	"encoding/hex"
 	"encoding/pem"
 	"fmt"
+	"io"
+	"net/http"
+	"strings"
 )
 
-// asserts/aws_nitro_enclaves.crt contains the PEM encoded roots for verifying Nitro
-//	Enclave attestation signatures. You can download them from
-//	https://aws-nitro-enclaves.amazonaws.com/AWS_NitroEnclaves_Root-G1.zip
-//	It's recommended you calculate the SHA256 sum of this string and match
-//	it to the one supplied in the AWS documentation
+// AWS_NitroEnclaves_Root-G1.zip file and its digest are advertised on
 //	https://docs.aws.amazon.com/enclaves/latest/user/verify-root.html
 
-//go:embed assets/aws_nitro_enclaves.crt
-var NitroCertPEM []byte
+const AWSNitroEnclavesRootURL = "https://aws-nitro-enclaves.amazonaws.com/AWS_NitroEnclaves_Root-G1.zip"
 
-//go:embed assets/selfsigned_cert.der
-var selfSignedCertDER []byte
+//go:embed assets/AWS_NitroEnclaves_Root-G1.zip
+var AWSNitroEnclavesRootZip []byte
+
+//go:embed assets/AWS_NitroEnclaves_Root-G1.sha256.hex
+var AWSNitroEnclavesRootSHA256Hex string
+
+type UnzipRootsFunc func(zipBytes []byte) (pemBytes []byte, err error)
+
+func UnzipRoots(zipBytes []byte) ([]byte, error) {
+	zipReader, err := zip.NewReader(
+		bytes.NewReader(zipBytes),
+		int64(len(zipBytes)),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating zip reader: %w", err)
+	}
+
+	if len(zipReader.File) != 1 {
+		return nil, fmt.Errorf("unexpected file count: %d", len(zipReader.File))
+	}
+	if zipReader.File[0].Name != "root.pem" {
```

Comment:
The `UnzipRoots` only needs to unzip the particular archive provider by AWS, which contains the file `root.pem` - all very specific stuff. I pull this function out, so that its easy to mock for testing.